### ISSUE TITLE
Override AppboyFirebaseMessagingService using a custom FirebaseService fails sending tokens etc to braze 

### ISSUE THREAD

**Issue by shanukp** [Appboy/appboy-unity-sdk#89]
Hi 

we have multiple sources from where we schedule PN .. one of them is braze .. to support this 
I created a custom FirebaseService extends com.google.firebase.messaging.FirebaseMessagingService
and then inside that

`@Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
     
        if (AppboyFirebaseMessagingService.isBrazePushNotification(remoteMessage)) {
            if (checkShowMode(this)) {
                Log.i(TAG,"Handling push notification with Braze");
                AppboyFirebaseMessagingService.handleBrazeRemoteMessage(this, remoteMessage);
            }
        } else {
                Log.i(TAG,"Handling others");
        }
    }`

and in Androidmanifest.xml replaced 
<service android:name="com.appboy.AppboyFirebaseMessagingService"
with

 ```
<service android:name="customFirebaseServicename">
      <intent-filter>
        <action android:name="com.google.firebase.MESSAGING_EVENT" />
      </intent-filter>
    </service>
```

After these changes , my firebase tokens generated are not getting updated to braze dashboard.

Please help

**Comment by shanukp** [comment]
doing adb logcat | grep "com.appboy.Appboy" for builds before and after this change as below

Before custom firebase class 
`08-01 05:19:13.390 30626 30626 I Braze v14.0.0 .com.appboy.Appboy: Appboy outbound network requests are now enabled
08-01 05:19:13.407 30626 30626 D Braze v14.0.0 .com.appboy.Appboy: Braze SDK loaded in 14 ms.
08-01 05:19:13.410 30626  2919 V Braze v14.0.0 .com.appboy.Appboy: Starting up a new user dependency manager
08-01 05:19:13.426 30626  2915 I Braze v14.0.0 .com.appboy.Appboy: Firebase Cloud Messaging found. Setting up Firebase Cloud Messaging.
08-01 05:19:13.428 30626  2915 W Braze v14.0.0 .com.appboy.Appboy: ADM manifest requirements not met. Braze will not register for ADM.`


After 

`08-01 05:20:05.338  3059  3059 I Braze v14.0.0 .com.appboy.Appboy: Appboy outbound network requests are now enabled
08-01 05:20:05.348  3059  3059 D Braze v14.0.0 .com.appboy.Appboy: Braze SDK loaded in 7 ms.
08-01 05:20:05.350  3059  5281 V Braze v14.0.0 .com.appboy.Appboy: Starting up a new user dependency manager
08-01 05:20:05.358  3059  5280 I Braze v14.0.0 .com.appboy.Appboy: Firebase Cloud Messaging found. Setting up Firebase Cloud Messaging.
08-01 05:20:05.374  3059  5280 W Braze v14.0.0 .com.appboy.Appboy: ADM manifest requirements not met. Braze will not register for ADM.
08-01 05:20:07.168  3059  5467 V Braze v14.0.0 .com.appboy.AppboyFirebaseMessagingService: Registering Firebase push token in onNewToken. Token: cXR-XaqyS0ioJmHQSSCkzX:APA91bGm_6YE1O3DUrtx_gvPzyVzdEHEuxVvzgUSnan2EBzPpktU2G2-4FSJb-A3St_CNDNSHqADlBT2_wcORu1ONjSMJJEEPb3Ubzrn5yYrzPsP74WQ-FeLod0Ch1mb00u9nZx4B5h9
08-01 05:20:07.168  3059  5467 I Braze v14.0.0 .com.appboy.Appboy: Push token cXR-XaqyS0ioJmHQSSCkzX:APA91bGm_6YE1O3DUrtx_gvPzyVzdEHEuxVvzgUSnan2EBzPpktU2G2-4FSJb-A3St_CNDNSHqADlBT2_wcORu1ONjSMJJEEPb3Ubzrn5yYrzPsP74WQ-FeLod0Ch1mb00u9nZx4B5h9 registered and immediately being flushed.
08-01 05:20:07.169  3059  5467 V Braze v14.0.0 .com.appboy.AppboyFirebaseMessagingService: Registering Firebase push token in onNewToken. Token: cXR-XaqyS0ioJmHQSSCkzX:APA91bGm_6YE1O3DUrtx_gvPzyVzdEHEuxVvzgUSnan2EBzPpktU2G2-4FSJb-A3St_CNDNSHqADlBT2_wcORu1ONjSMJJEEPb3Ubzrn5yYrzPsP74WQ-FeLod0Ch1mb00u9nZx4B5h9
08-01 05:20:07.169  3059  5467 I Braze v14.0.0 .com.appboy.Appboy: Push token cXR-XaqyS0ioJmHQSSCkzX:APA91bGm_6YE1O3DUrtx_gvPzyVzdEHEuxVvzgUSnan2EBzPpktU2G2-4FSJb-A3St_CNDNSHqADlBT2_wcORu1ONjSMJJEEPb3Ubzrn5yYrzPsP74WQ-FeLod0Ch1mb00u9nZx4B5h9 registered and immediately being flushed.
^C`



Basically looks like _Registering Firebase push token in onNewToken._ - this flow is not getting triggered for me after the custom changes

**Comment by radixdev** [comment]
We need verbose logcat output, your custom endpoint, and internal dashboard links to your user to properly debug what you're seeing. Additionally, I don't think this is a bug in the Braze Unity SDK itself. As such, I would highly recommend emailing support@braze.com for assistance.
